plugins/kubernetes/src/utils/clusterLinks/formatters/openshift.ts

```diff
@@ -15,10 +15,30 @@
  */
 import { ClusterLinksFormatterOptions } from '../../../types/types';
 
-export function openshiftFormatter(
-  _options: ClusterLinksFormatterOptions,
-): URL {
-  throw new Error(
-    'OpenShift formatter is not yet implemented. Please, contribute!',
-  );
+const kindMappings: Record<string, string> = {
+  deployment: 'deployments',
+  ingress: 'ingresses',
+  service: 'services',
+  horizontalpodautoscaler: 'horizontalpodautoscalers',
+  persistentvolume: 'persistentvolumes',
+};
+
+export function openshiftFormatter(options: ClusterLinksFormatterOptions): URL {
+  const result = new URL(options.dashboardUrl.href);
+  const name = options.object.metadata?.name;
+  const namespace = options.object.metadata?.namespace;
+  const validKind = kindMappings[options.kind.toLocaleLowerCase('en-US')];
+  if (namespace) {
+    if (name && validKind) {
+      result.pathname = `k8s/ns/${namespace}/${validKind}/${name}`;
```

Comment:
Would you mind adding an `encodeURIComponent(x)` on all dynamic path parts? I know that it's common to only permit URL safe chars for these parts, but Backstage permits users to tweak those rules. And either way - it's good practice to not trust user input :)

Reply:
Not if they contain slashes ;) plus that auto quoting behavior is really odd - it tries its best to avoid double encoding etc using some heuristic 🤔

Reply:
ok, fine, I will do the encoding in all formatters...
I thought that forcing the URL type would solve the issue of ignoring the encoding, but if it doesn't, I might revert to the String type...

Reply:
I did the refactoring and the encoding